app/src/lib/download.ts

```diff
@@ -0,0 +1,25 @@
+import {getAuthHeader} from "./auth/getAuthHeader";
+
+export const download = async (url: string, filename: string, authRequired = true) => {
+    const headers = authRequired ? getAuthHeader(): undefined;
+    const res = await fetch(url, {
+        method: "GET",
+        headers,
+    });
+
+    if (!res.ok) {
+        const json = await res.json();
+        const msg = json.error?.detail ? `Error: ${json.error.detail}` : `Error downloading ${filename}`;
+        throw new Error(msg);
+    }
+
+    const blob = await res.blob();
+    const fileUrl = URL.createObjectURL(blob);
```

Comment:
sorry just my ignorance but is this the only way to do this? seems really hacky for some reason

Reply:
It seems to be sadly... Yeah, that was my feeling too, but this seems to be the standard approach.